user.go

```diff
@@ -2,414 +2,449 @@ package passage
 
 import (
 	"context"
-	"fmt"
+	"encoding/json"
 	"net/http"
 	"strings"
 )
 
-// GetUser gets a user using their userID
-// returns user on success, error on failure
-//
-// Deprecated: Use Passage.User.Get() instead.
-func (a *App) GetUser(userID string) (*User, error) {
-	res, err := a.client.GetUserWithResponse(context.Background(), a.ID, userID)
+type PassageUser = User
+type CreateUserArgs = CreateUserBody
+type UpdateUserOptions = UpdateBody
+
+type user struct {
+	appID  string
+	client *ClientWithResponses
+}
+
+func newUser(appID string, client *ClientWithResponses) *user {
+	return &user{
+		appID:  appID,
+		client: client,
+	}
+}
+
+// Get retrieves a user's object using their user ID.
+func (u *user) Get(userID string) (*PassageUser, error) {
+	res, err := u.client.GetUserWithResponse(context.Background(), u.appID, userID)
 	if err != nil {
-		return nil, networkError()
+		return nil, err
 	}
 
 	if res.JSON200 != nil {
 		return &res.JSON200.User, nil
 	}
 
-	var errorText string
+	var message string
 	var errorCode string
 	switch {
 	case res.JSON401 != nil:
-		errorText = res.JSON401.Error
+		message = res.JSON401.Error
 		errorCode = string(res.JSON401.Code)
 	case res.JSON404 != nil:
-		errorText = res.JSON404.Error
+		message = res.JSON404.Error
 		errorCode = string(res.JSON404.Code)
 	case res.JSON500 != nil:
-		errorText = res.JSON500.Error
+		message = res.JSON500.Error
 		errorCode = string(res.JSON500.Code)
+	default:
+		var errorBody httpErrorBody
+		if err := json.Unmarshal(res.Body, &errorBody); err != nil {
+			return nil, err
+		}
+
+		message = errorBody.Error
+		errorCode = errorBody.Code
 	}
 
-	return nil, Error{
-		Message:    errorText,
-		StatusCode: res.StatusCode(),
-		StatusText: res.Status(),
-		ErrorText:  errorText,
+	return nil, PassageError{
+		Message:    message,
 		ErrorCode:  errorCode,
+		StatusCode: res.StatusCode(),
 	}
```

Comment:
i want to solve all of this repeated error handling code with generics (since each generated response type is different per function), but it requires go >= 1.18.

i will address it in another PR that updates the toolchain for the minor release.